### PR TITLE
Readme overhaul

### PR DESCRIPTION
**Pull request by mike1729**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,25 +1,37 @@
...
+readme = "README.md"
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+description = """
+    Aleph is an asynchronous and Byzantine fault tolerant consensus protocol aimed
+    at ordering arbitrary messages (transactions). It has been designed to operate
+    continuously under conditions where there is no bound on message-delivery delay
+    and under the assumption that there is a significant probability of malicious
+    behavior, making it an excellent fit for blockchain-related applications.
+    """
```
Comment (DamianStraszak) [inline Cargo.toml:20]: It has been designed to continuously operate even in the harshest conditions: with no bounds on message-delivery delays and in the presence of malicious actors. This makes it an excellent fir for blockchain-related applications.

File: README.md [hunk 2]
```diff
@@ -1,19 +1,153 @@
...
+If the crate's [documentation][docs-link] seems to be not comprehensive enough,
+please refer to the [detailed version][reference-link].
+
+### Implementation status
+
+- current version is asynchronous, so it's driven by consensus events as opposed
+  to some clock ticks
+- while being asynchronous, the performance is still optimal in partially
+  synchronous environment
+- guaranteed safety in partially synchronous environment
```
Comment (DamianStraszak) [inline README.md:36]: - guaranteed safety even in asynchronous environment

File: README.md [hunk 3]
```diff
@@ -1,19 +1,153 @@
...
+- secure against fork bombs, for details see [the paper][paper-link]
+- network overhead optimized to not send all parents hashes but a bitmap and a control hash
+- thorough testing, including malicious scenarios, and high code coverage
+
+### Future work
+
+- asynchronous liveness - current version is not secure in a rather theoretic
+  scenario when an adversary controls all the delays in the network, including
+  honest nodes messages. However, we take security very seriously, so we plan to
+  make the consensus secure even in such a unusual scenario.
```
Comment (DamianStraszak) [inline README.md:47]: asynchronous liveness - the current version is not live under full asynchrony. It is unclear whether this has any practical relevance, but we are making efforts to strenghen the guarantees in this direction. Adding a safe source of randomness to the protocol allows to make it fully asynchronous.

File: Cargo.toml (hunk 1 above)
Comment (mike1729) [reply Cargo.toml:20]: Changed.

File: README.md (hunk 2 above)
Comment (mike1729) [reply README.md:36]: Changed.

File: README.md (hunk 3 above)
Comment (mike1729) [reply README.md:47]: Changed.

File: README.md [hunk 4]
```diff
@@ -1,19 +1,160 @@
...
+- Import AlephBFT in your crate
+  ```
+  [dependencies]:
+  AlephBFT = "1"
+  ```
+- AlephBFT requires user to provide it with an implementation of the following traits:
+  - The [DataIO][dataio-link] trait is an abstraction for a component that provides data items,
+    checks availability of data items and allows to input ordered data items. `DataIO` is
+    parametrized with a `Data` generic type representing the type of items we would like to order.
+    ```
```
Comment (kostekIV) [inline README.md:68]: Adding `rust` here may colour the syntax of examples

Comment (mike1729) [reply README.md]: That's neat! done.

**Pull request merged**